src/work/PokerGame/PokerGame.sol

```diff
@@ -0,0 +1,516 @@
+        uint256 randomness = randomWords[0];
+        shuffleDeck(gameId, randomness);
+        delete vrfRequests[requestId];
+        continueGameSetup(gameId);
+    }
+
+    // Modified shuffleDeck function to use Chainlink VRF
+    function shuffleDeck(uint256 gameId, uint256 randomness) internal {
+        Deck storage deck = decks[gameId];
+        for (uint8 i = 51; i > 0; i--) {
+            // Use the provided randomness to generate a random index
+            uint8 j = uint8(uint256(keccak256(abi.encodePacked(randomness, i))) % (i + 1));
+            // Swap the current card with the randomly selected card
+            Card memory temp = deck.cards[i];
+            deck.cards[i] = deck.cards[j];
+            deck.cards[j] = temp;
+        }
+        // Emit an event to signal that the deck has been shuffled
+        emit DeckShuffled(gameId);
+    }
+
+    // Function to start the game
+    function startGame(uint256 gameId) public onlyGameCreator(gameId) {
+        Game storage game = games[gameId]; // Get the game from storage
+        require(game.state == GameState.WAITING, "Game not in waiting state"); // Ensure the game is in the WAITING state
+        require(game.playerCount >= 2, "Not enough players"); // Ensure there are at least 2 players
+
+        initializeDeck(gameId);  // Initialize the deck
+        game.state = GameState.SHUFFLING; // New state to indicate waiting for VRF
+        requestRandomness(gameId);
+        emit GameShuffling(gameId);
+    }
+
+    // New function to continue game setup after shuffling
+    function continueGameSetup(uint256 gameId) internal {
+        Game storage game = games[gameId];
+        require(game.state == GameState.SHUFFLING, "Game not in shuffling state");
+
+        game.state = GameState.ACTIVE;
+        game.dealerIndex = 0;
+        game.smallBlindIndex = (game.dealerIndex + 1) % game.playerCount;
+        game.bigBlindIndex = (game.smallBlindIndex + 1) % game.playerCount;
+        game.currentPlayerIndex = (game.bigBlindIndex + 1) % game.playerCount;
+
+        address firstPlayer = game.players[game.currentPlayerIndex].addr;
+        _grantRole(CURRENT_TURN_ROLE, firstPlayer);
+        game.players[game.currentPlayerIndex].hasCurrentTurnRole = true;
+
+        collectBlind(gameId, game.smallBlindIndex, 1);
+        collectBlind(gameId, game.bigBlindIndex, 2);
+
+        emit GameStarted(gameId);
+        emit TurnChanged(gameId, game.players[game.currentPlayerIndex].addr);
+    }
+
+    // Internal function to collect blinds from a player
+    function collectBlind(uint256 gameId, uint8 playerIndex, uint256 blindAmount) internal {
+        Game storage game = games[gameId]; // Get the game from storage
+        Player storage player = game.players[playerIndex]; // Get the player from the game
+
+        require(player.chipCount >= blindAmount, "Player doesn't have enough chips for blind"); // Ensure the player has enough chips
```

Comment:
This results in the vrf callback reverting, this isnt the best because it's reverting a 3rd party method call. I think its better to have the separate from the callback and make it external so you can call it yourself again when ready.